### PR TITLE
Fix autoimport raising AttributeError in some cases

### LINKED ISSUE #307: Autoimport keep throwing exception when delete a line
```
--- Send (notification): textDocument/didChange

--- Send (58706): textDocument/completion
2022-11-28 22:45:51,446 CST - WARNING - pylsp.config.config - Failed to load hook pylsp_completions: 'NoneType' object has no attribute 'value'
Traceback (most recent call last):
  File "▒▒▒▒▒▒▒▒/venv/lib/python3.10/site-packages/pylsp/config/config.py", line 33, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "▒▒▒▒▒▒▒▒/venv/lib/python3.10/site-packages/pluggy/_manager.py", line 327, in traced_hookexec
    return outcome.get_result()
  File "▒▒▒▒▒▒▒▒/venv/lib/python3.10/site-packages/pluggy/_result.py", line 60, in get_result
    raise ex[1].with_traceback(ex[2])
  File "▒▒▒▒▒▒▒▒/venv/lib/python3.10/site-packages/pluggy/_result.py", line 33, in from_call
    result = func()
  File "▒▒▒▒▒▒▒▒/venv/lib/python3.10/site-packages/pluggy/_manager.py", line 324, in <lambda>
    lambda: oldcall(hook_name, hook_impls, kwargs, firstresult)
  File "▒▒▒▒▒▒▒▒/venv/lib/python3.10/site-packages/pluggy/_callers.py", line 60, in _multicall
    return outcome.get_result()
  File "▒▒▒▒▒▒▒▒/venv/lib/python3.10/site-packages/pluggy/_result.py", line 60, in get_result
    raise ex[1].with_traceback(ex[2])
  File "▒▒▒▒▒▒▒▒/venv/lib/python3.10/site-packages/pluggy/_callers.py", line 39, in _multicall
    res = hook_impl.function(*args)
  File "▒▒▒▒▒▒▒▒/venv/lib/python3.10/site-packages/pylsp/plugins/rope_autoimport.py", line 159, in pylsp_completions
    word = word_node.value
AttributeError: 'NoneType' object has no attribute 'value'
```

After I digging in code, I found the var [`line`](https://github.com/python-lsp/python-lsp-server/blob/develop/pylsp/plugins/rope_autoimport.py#L154) is alway a empty string when I delete a line

### PR DESCRIPTION
**Pull request by douo**
More detail in #307 

Fixes #307

### CODE REVIEW

**Review by ccordoba12 — COMMENTED** [review]
Thanks for your contribution @douo! To fix the linting error reported by Github actions, please add the following inline comment next to the definition of `_should_insert`:

```python
def _should_insert(expr: tree.BaseNode, word_node: tree.Leaf) -> bool:  # pylint: disable=too-many-return-statements
```

Otherwise looks good to me.

**Comment by douo** [comment]
Done, Thanks @ccordoba12.

File: test/plugins/test_autoimport.py [hunk 1]
```diff
@@ -100,6 +100,9 @@ def test_autoimport_function(completions):
 def test_autoimport_class(completions):
     assert len(completions) == 0
 
+@pytest.mark.parametrize("completions", [("""\n""", 0)], indirect=True)
+def test_autoimport_empty_line(completions):
+    assert len(completions) == 0
```
Comment (ccordoba12) [inline test/plugins/test_autoimport.py:106]:
```suggestion

@pytest.mark.parametrize("completions", [("""\n""", 0)], indirect=True)
def test_autoimport_empty_line(completions):
    assert len(completions) == 0

```

**Review by ccordoba12 — COMMENTED** [review]
Please apply the suggestion I left below to please another of our linters.

**Review by ccordoba12 — APPROVED** [review]
Thanks a lot for your help @douo! We really appreciate it.

**Pull request merged**